pkg/binpack/single_az_pack_tightly_test.go

```diff
@@ -88,19 +88,37 @@ func TestSingleAZTightlyPack(t *testing.T) {
 		nodePriorityOrder:  []string{"n1_z1", "n1_z2"},
 		expectedDriverNode: "n1_z1",
 		willFit:            false,
+	}, {
+		name:              "prefer a result with better packing for selected nodes",
+		driverResources:   createResources(1, 0, 0),
+		executorResources: createResources(0, 0, 0),
+		numExecutors:      0,
+		nodesSchedulingMetadata: resources.NodeGroupSchedulingMetadata(map[string]*resources.NodeSchedulingMetadata{
+			"n1_z1": createSchedulingMetadataWithTotals(0, 10, 0, 0, 0, 0, "z1"),
+			"n2_z1": createSchedulingMetadataWithTotals(0, 10, 0, 0, 0, 0, "z1"),
+			"n3_z1": createSchedulingMetadataWithTotals(10, 10, 0, 0, 0, 0, "z1"),
+			"n1_z2": createSchedulingMetadataWithTotals(9, 10, 0, 0, 0, 0, "z2"),
+			"n2_z2": createSchedulingMetadataWithTotals(10, 10, 0, 0, 0, 0, "z2"),
+			"n3_z2": createSchedulingMetadataWithTotals(10, 10, 0, 0, 0, 0, "z2"),
+		}),
+		nodePriorityOrder:      []string{"n1_z1", "n2_z1", "n3_z1", "n1_z2", "n2_z2", "n3_z2"},
+		expectedDriverNode:     "n1_z2",
+		willFit:                true,
+		expectedExecutorCounts: map[string]int{},
```

Comment:
Can we have a few more larger examples with executors to make sure we're covering all our bases here

Reply:
I agree. Adding a couple of test cases.

Reply:
Added a carefully chosen corner case test. AZ1 yields better average packing efficiency across AZ and across all nodes. AZ2 yields better average packing efficiency over 2 nodes in AZ2 but otherwise worse.

Reply:
Nice